firmware/led-spiout.c

```diff
@@ -192,64 +185,128 @@ void led_set_spi_frequency(uint8_t frequency) {
     }
 }
 
+/* Sets all leds off, without using the interrupt */
+static void _led_all_off_no_interrupt() {
+    /* make sure we disabled the interrupt ! */
+    DISABLE_LED_WRITES;
 
-void led_init() {
+#define WAIT_SPI_TRANSMIT()     do { while (!(SPSR & _BV(SPIF))) ; } while (0)
+    SPDR = 0;
```

Comment:
Since I think we want to just be writing all zeros, we ought to be able to have a single for loop, with I less than 

(LED_START_FRAME_BYTES+ (NUM_LEDS*4) + END_FRAME_BYTES)

Reply:
(Not if `BRIGHTNESS_MASK is actually a `LED_SPI_BEGIN_DATA_BITS`)